pyprep/find_noisy_channels.py

```diff
@@ -92,65 +96,73 @@ def __init__(self, raw, do_detrend=True, random_state=None, matlab_strict=False)
+        """Get a list of all channels currently flagged as bad.
 
-        This function makes a list of all the bad channels and prints them if verbose
-        is True.
+        Note that this method does not perform any bad channel detection itself,
+        and only reports channels already detected as bad by other methods.
 
         Parameters
         ----------
         verbose : bool
-            If verbose, print a summary of bad channels.
+            If ``True``, a summary of the channels currently flagged as by bad per
+            category is printed. Defaults to ``False``.
+
+        Returns
+        -------
+        bads : list
+            THe names of all bad channels detected by any method so far.
 
         """
-        bads = (
-            self.bad_by_nan
-            + self.bad_by_flat
-            + self.bad_by_deviation
-            + self.bad_by_hf_noise
-            + self.bad_by_SNR
-            + self.bad_by_correlation
-            + self.bad_by_dropout
-            + self.bad_by_ransac
-        )
-        bads = list(set(bads))
+        bads = {
+            "n/a": self.bad_by_nan,
+            "flat": self.bad_by_flat,
+            "deviation": self.bad_by_deviation,
+            "hf noise": self.bad_by_hf_noise,
+            "correl": self.bad_by_correlation,
+            "SNR": self.bad_by_SNR,
+            "dropout": self.bad_by_dropout,
+            "RANSAC": self.bad_by_ransac,
+        }
+
+        all_bads = set()
+        for bad_chs in bads.values():
+            all_bads.update(bad_chs)
 
         if verbose:
-            print("Found {} uniquely bad channels.".format(len(bads)))
-            print("\n{} by n/a: {}".format(len(self.bad_by_nan), self.bad_by_nan))
-            print("\n{} by flat: {}".format(len(self.bad_by_flat), self.bad_by_flat))
-            print(
-                "\n{} by deviation: {}".format(
-                    len(self.bad_by_deviation), self.bad_by_deviation
-                )
-            )
-            print(
-                "\n{} by hf noise: {}".format(
-                    len(self.bad_by_hf_noise), self.bad_by_hf_noise
-                )
-            )
-            print(
-                "\n{} by correl: {}".format(
-                    len(self.bad_by_correlation), self.bad_by_correlation
-                )
-            )
-            print("\n{} by SNR {}".format(len(self.bad_by_SNR), self.bad_by_SNR))
-            print(
-                "\n{} by dropout: {}".format(
-                    len(self.bad_by_dropout), self.bad_by_dropout
-                )
-            )
-            print(
-                "\n{} by ransac: {}".format(len(self.bad_by_ransac), self.bad_by_ransac)
-            )
-        return bads
+            out = "Found {} uniquely bad channels:\n".format(len(all_bads))
+            for bad_type, bad_chs in bads.items():
+                out += "\n{} by {}: {}\n".format(len(bad_chs), bad_type, bad_chs)
+            print(out)
```

Comment:
I didn't actually know that was a thing! Most of my early Python experience was writing code that had to be 2.7-compatible, so I've slept on some of the newer language features. Since 3.7 is the minimum here I'll update to this style since I agree that it's cleaner!